<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>804</width>
    <height>453</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Simplifying Poligon</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="1" column="0">
     <widget class="Display" name="display" native="true"/>
    </item>
    <item row="0" column="0">
     <layout class="QGridLayout" name="controlBar">
      <item row="3" column="3" colspan="3">
       <widget class="QSlider" name="horizontalSlider2">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>50</width>
          <height>0</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickInterval">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QCheckBox" name="checkBox1">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="whatsThis">
         <string/>
        </property>
        <property name="text">
         <string>Ramer-Douglas-Peucker</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="3" colspan="3">
       <widget class="QSlider" name="horizontalSlider1">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>50</width>
          <height>0</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="7">
       <widget class="QLabel" name="label11">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>2</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QCheckBox" name="checkBox2">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="text">
         <string>Visvalingam-Whyatt</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="8">
       <widget class="QLabel" name="label12">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Min. tolerancija:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="6">
       <widget class="QLabel" name="label20">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Broj tačaka:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="9">
       <widget class="QLabel" name="label13">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>50</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>0.00000</string>
        </property>
       </widget>
      </item>
      <item row="3" column="8">
       <widget class="QLabel" name="label22">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Sledeća promena:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="6">
       <widget class="QLabel" name="label10">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Broj tačaka:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="9">
       <widget class="QLabel" name="label23">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>50</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>0.00000</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="checkBox01">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="text">
         <string>Temena</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="7">
       <widget class="QLabel" name="label21">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>3</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="checkBox0">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="text">
         <string>Ceo poligon</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="10">
       <layout class="QHBoxLayout" name="horizontalLayoutInput">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <widget class="QPushButton" name="pushButtonHelp">
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
          <property name="text">
           <string>Uputstvo</string>
          </property>
          <property name="checkable">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_5">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label0">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Unos tačaka poligona:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_4">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonUnos1">
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
          <property name="text">
           <string>Učitaj iz fajla</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinBoxRandom">
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="maximum">
           <number>500</number>
          </property>
          <property name="value">
           <number>3</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonUnos2">
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
          <property name="text">
           <string>Generiši nasumično</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonUnos30">
          <property name="minimumSize">
           <size>
            <width>95</width>
            <height>0</height>
           </size>
          </property>
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
          <property name="text">
           <string>Interaktivan unos</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelX">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>X</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEditX">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelY">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Y</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEditY">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonUnos31">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Dodaj Teme</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>2</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="1" column="0" colspan="10">
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="checkBoxAA">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="text">
         <string>AntiAliasing</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QPushButton" name="pushButtonColor">
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="text">
         <string>Boje</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Display</class>
   <extends>QWidget</extends>
   <header location="global">display.h</header>
   <container>1</container>
   <slots>
    <signal>changeLimit1(double)</signal>
    <signal>updateSliderValues(int)</signal>
    <signal>setMinTolerance(double)</signal>
    <signal>setLastAreaChange(double)</signal>
    <slot>displayStartPolygon(bool)</slot>
    <slot>displayAlg1Polygon(bool)</slot>
    <slot>displayAlg2Polygon(bool)</slot>
    <slot>setAlg1Points(int)</slot>
    <slot>setAlg2Points(int)</slot>
    <slot>readFromFile()</slot>
    <slot>setRandomPoints(int)</slot>
    <slot>generateRandom()</slot>
    <slot>displayPoints(bool)</slot>
    <slot>changeColors()</slot>
    <slot>toggleAA(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>checkBox2</tabstop>
  <tabstop>checkBox0</tabstop>
  <tabstop>checkBox01</tabstop>
  <tabstop>horizontalSlider2</tabstop>
  <tabstop>pushButtonUnos1</tabstop>
  <tabstop>spinBoxRandom</tabstop>
  <tabstop>lineEditX</tabstop>
  <tabstop>lineEditY</tabstop>
  <tabstop>pushButtonUnos31</tabstop>
  <tabstop>pushButtonUnos2</tabstop>
  <tabstop>pushButtonUnos30</tabstop>
  <tabstop>horizontalSlider1</tabstop>
  <tabstop>checkBox1</tabstop>
  <tabstop>pushButtonHelp</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pushButtonUnos1</sender>
   <signal>clicked()</signal>
   <receiver>display</receiver>
   <slot>readFromFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>142</x>
     <y>339</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>display</receiver>
   <slot>setAlg1Points(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>501</x>
     <y>193</y>
    </hint>
    <hint type="destinationlabel">
     <x>287</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>display</receiver>
   <slot>setAlg2Points(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>398</x>
     <y>206</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>303</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox1</sender>
   <signal>clicked(bool)</signal>
   <receiver>display</receiver>
   <slot>displayAlg1Polygon(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>313</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox2</sender>
   <signal>clicked(bool)</signal>
   <receiver>display</receiver>
   <slot>displayAlg2Polygon(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>313</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox0</sender>
   <signal>clicked(bool)</signal>
   <receiver>display</receiver>
   <slot>displayStartPolygon(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>146</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>82</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxRandom</sender>
   <signal>valueChanged(int)</signal>
   <receiver>display</receiver>
   <slot>setRandomPoints(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>354</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>325</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonUnos2</sender>
   <signal>clicked()</signal>
   <receiver>display</receiver>
   <slot>generateRandom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>406</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>304</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>display</sender>
   <signal>updateSliderValues(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setTickMarks(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>439</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setLabel1(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>398</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>407</x>
     <y>99</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setLabel2(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>398</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>447</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>display</sender>
   <signal>setMinTolerance(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setLabel13(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>541</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>559</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>display</sender>
   <signal>setLastAreaChange(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setLabel23(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>596</x>
     <y>251</y>
    </hint>
    <hint type="destinationlabel">
     <x>675</x>
     <y>215</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox01</sender>
   <signal>clicked(bool)</signal>
   <receiver>display</receiver>
   <slot>displayPoints(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>52</x>
     <y>251</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonUnos30</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>interactiveInputModeChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>516</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>389</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonUnos31</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>interactiveInputPassPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>786</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>715</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonHelp</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>showAboutWindow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>53</x>
     <y>81</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonColor</sender>
   <signal>clicked()</signal>
   <receiver>display</receiver>
   <slot>changeColors()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>41</x>
     <y>208</y>
    </hint>
    <hint type="destinationlabel">
     <x>23</x>
     <y>258</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxAA</sender>
   <signal>clicked(bool)</signal>
   <receiver>display</receiver>
   <slot>toggleAA(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>42</x>
     <y>178</y>
    </hint>
    <hint type="destinationlabel">
     <x>37</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setTickMarks(int)</slot>
  <slot>setLabel1(int)</slot>
  <slot>setLabel2(int)</slot>
  <slot>readFromFile()</slot>
  <slot>setLabel13(double)</slot>
  <slot>setLabel23(double)</slot>
  <slot>interactiveInputModeChange()</slot>
  <slot>interactiveInputPassPoint()</slot>
  <slot>showAboutWindow()</slot>
  <slot>colorDialog()</slot>
  <slot>toggleAA(bool)</slot>
 </slots>
</ui>
